homeassistant/components/lightwave/climate.py

```diff
@@ -0,0 +1,167 @@
+"""Support for LightwaveRF TRVs."""
+import logging
+
+from homeassistant.components.climate import (
+    DEFAULT_MAX_TEMP,
+    DEFAULT_MIN_TEMP,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+    ClimateDevice,
+)
+from homeassistant.components.climate.const import CURRENT_HVAC_HEAT, CURRENT_HVAC_OFF
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL,
+    ATTR_TEMPERATURE,
+    CONF_NAME,
+    TEMP_CELSIUS,
+)
+
+from . import CONF_SERIAL, LIGHTWAVE_LINK
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Find and return LightWave lights."""
+    if not discovery_info:
+        return
+
+    trv = []
+    lwlink = hass.data[LIGHTWAVE_LINK]
+
+    for device_id, device_config in discovery_info.items():
+        name = device_config[CONF_NAME]
+        serial = device_config[CONF_SERIAL]
+        trv.append(LightwaveTrv(name, device_id, lwlink, serial))
+
+    async_add_entities(trv)
+
+
+class LightwaveTrv(ClimateDevice):
+    """Representation of a LightWaveRF TRV."""
+
+    def __init__(self, name, device_id, lwlink, serial):
+        """Initialize LightwaveTrv entity."""
+        self._name = name
+        self._device_id = device_id
+        self._state = None
+        self._max_temp = DEFAULT_MAX_TEMP
+        self._min_temp = DEFAULT_MIN_TEMP
+        self._current_temperature = None
+        self._target_temperature = None
+        self._temperature_unit = TEMP_CELSIUS
+        self._target_temperature_step = 0.5
+        self._hvac_mode = HVAC_MODE_HEAT
+        self._hvac_action = None
+        self._lwlink = lwlink
+        self._battery = None
+        self._serial = serial
+        # inhibit is used to prevent race condition on update.  If non zero, skip next update cycle.
+        self._inhibit = 0
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_TARGET_TEMPERATURE
+
+    def update(self):
+        """Communicate with a Lightwave RTF Proxy to get state."""
+        targ = temp = battery = trv_output = None
+        (temp, targ, battery, trv_output) = self._lwlink.read_trv_status(self._serial)
```

Comment:
Can each individual value be `None` or wouldn't all values be `None` if reading the trv status fails?
Because if all values are automatically None if the reading fails you could combine the following multiple if-statements into one.

Reply:
The proxy is external to HA, so I cannot be certain of that behaviour.
While it is true today, I can't be sure that someone develops a different proxy with different behaviour.   So suggest keeping as is.